File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} candidates
     * @param {number} target
     * @return {number[][]}
     */
    combinationSum2(candidates, target) {
        candidates.sort((a,b) => a-b);
        const res = [];

        dfs([], 0, 0);

        return res;

        function dfs(tmp, sum, idx) {
            if (sum > target) return;
            if (sum === target) {
                res.push([...tmp]);
                return;
            }

            for (let i = idx; i < candidates.length; i++) {
                if (i > idx && candidates[i] === candidates[i-1]) continue;
                tmp.push(candidates[i]);
                dfs(tmp, sum+candidates[i], i+1);
                tmp.pop();
            }
        }
    }
}

File: Data Structures & Algorithms/count-number-of-islands/submission-1.go
func numIslands(grid [][]byte) int {
    m, n := len(grid), len(grid[0])
    res := 0

    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if grid[i][j] == byte('1') {
                dfs(grid, i, j)
                res++
            }
        }
    }

    return res
}

func dfs(grid [][]byte, i, j int) {
    m, n := len(grid), len(grid[0])
    if i < 0 || i >= m || j < 0 || j >= n || grid[i][j] != byte('1') {
        return
    }

    grid[i][j] = byte('0')

    dfs(grid, i+1, j)
    dfs(grid, i-1, j)
    dfs(grid, i, j+1)
    dfs(grid, i, j-1)
}
